pkg/virtctl/create/vm/vm.go

```diff
@@ -327,13 +327,13 @@ func (c *createVM) run(cmd *cobra.Command, _ []string) error {
 
 	for _, flag := range flags {
 		if cmd.Flags().Changed(flag) {
-			if err := optFns[flag](c, vm); err != nil {
+			if err := c.optFns()[flag](vm); err != nil {
```

Comment:
You are right. Depending on which flag is modified the loop wold need to run again. Because of the fixed order of execution of flags trough `flags`, we can get away with it right now. `--access-cred` only changes flags that relate to cloud-init for which the generation handler is executed after the handler of `--access-cred`.

Reply:
The same applies to the flags `--volume-{clone-pvc,datasource,blank}` in https://github.com/kubevirt/kubevirt/pull/13019.

Reply:
Ok, I see. I think a comment about this dependency would be useful, so we don't accidentally modify a flag our of order in the future. But I'm not sure where is the best place to put this comment.

Reply:
I'll add a comment above the `flags` slice.